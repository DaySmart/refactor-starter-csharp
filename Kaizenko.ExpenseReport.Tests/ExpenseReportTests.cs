using ApprovalTests.Reporters;
using NUnit.Framework;
using System;
using System.IO;
using System.Collections.Generic;

namespace Kaizenko.ExpenseReport.Tests
{
    public class ExpenseReportTests
    {
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void PrintReport_WhenNoExpensesProvided_ExpectAnEmptyReport()
        {
            using (StringWriter sw = new StringWriter())
            {
                Console.SetOut(sw);
                ExpenseReport expenseReport = new MyExpenseReport();
                expenseReport.PrintReport(new System.Collections.Generic.List<Expense>());
                ApprovalTests.Approvals.Verify(sw);
            }
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void PrintReport_Legacy()
        {
            using (StringWriter sw = new StringWriter())
            {
                
                Console.SetOut(sw);
                ExpenseReport expenseReport = new MyExpenseReport();
                expenseReport.PrintReport(getList());
                ApprovalTests.Approvals.Verify(sw);
            }
        }

        private List<Expense> getList()
        {
            List<Expense> expenses = new List<Expense>();
            ExpenseType[] expenseTypes = { ExpenseType.DINNER, ExpenseType.BREAKFAST, ExpenseType.CAR_RENTAL };
            foreach (ExpenseType expenseType in expenseTypes)
                for (int amount = 500; amount <= 8000; amount += 500)
                    expenses.Add(new Expense()
                    {
                        type = expenseType,
                        amount = amount
                    });
            return expenses;
        }
    }


    public class MyExpenseReport : ExpenseReport
    {
        protected override DateTime GetDate()
        {
            return new DateTime(2018, 08, 23, 08, 12, 59);
        }
    }
}
