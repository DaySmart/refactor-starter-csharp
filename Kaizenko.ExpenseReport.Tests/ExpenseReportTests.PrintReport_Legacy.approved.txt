Expenses 08/23/2018 08:12:59
Dinner	500	 
Dinner	1000	 
Dinner	1500	 
Dinner	2000	 
Dinner	2500	 
Dinner	3000	 
Dinner	3500	 
Dinner	4000	 
Dinner	4500	 
Dinner	5000	 
Dinner	5500	X
Dinner	6000	X
Dinner	6500	X
Dinner	7000	X
Dinner	7500	X
Dinner	8000	X
Breakfast	500	 
Breakfast	1000	 
Breakfast	1500	X
Breakfast	2000	X
Breakfast	2500	X
Breakfast	3000	X
Breakfast	3500	X
Breakfast	4000	X
Breakfast	4500	X
Breakfast	5000	X
Breakfast	5500	X
Breakfast	6000	X
Breakfast	6500	X
Breakfast	7000	X
Breakfast	7500	X
Breakfast	8000	X
Car Rental	500	 
Car Rental	1000	 
Car Rental	1500	 
Car Rental	2000	 
Car Rental	2500	 
Car Rental	3000	 
Car Rental	3500	 
Car Rental	4000	 
Car Rental	4500	 
Car Rental	5000	 
Car Rental	5500	 
Car Rental	6000	 
Car Rental	6500	 
Car Rental	7000	 
Car Rental	7500	 
Car Rental	8000	 
Meal expenses: 136000
Total expenses: 204000
